src/components/price/priceSaga.js

```diff
@@ -0,0 +1,17 @@
+import { call, put, takeLatest } from "redux-saga/effects";
+import { FETCH_PRICE_SUCCESS } from "./priceActions";
+import { SELECTED_SYMBOL } from "../search/searchActions"
+import { api } from "../../utils/apiUtil";
+
+/**** Workers ****/
+
+function* priceWorker({payload}) {
+  const price = yield call(api.getPrice, payload);
+  yield put({ type: FETCH_PRICE_SUCCESS, payload: price });
+}
+
+/**** Watchers ****/
+
+export function* priceWatcher() {
+  yield takeLatest(SELECTED_SYMBOL, priceWorker);
+}
```

Comment:
This seems to be a very repetitive pattern. Perhaps it would make sense to create a custom `selectedSymbol` effect, something like:

```js
const selectedSymbolSaga = (apiFn, actionType) => takeLatest(
  SELECTED_SYMBOL,
  function* ({payload}) {
    const response = yield call(apiFn, payload);
    yield put({ type: actionType, payload: response });
  }
);
```

And then even in the `rootSaga` we could do this:

```js
const selectedSymbolEffects = [
  [api.getPrice, FETCH_PRICE_SUCCESS],
  [api.getPeers, FETCH_PEER_SUCCESS],
  // etc
].map(args => selectedSymbolSaga(...args));

export function* root() {
  yield all(selectedSymbolEffects);
  yield fork(chartSaga);
  yield put(onAppStart());
}
```
This would make even more sense if we were using the ducks pattern and we had our "ducks" separated from the components... Then we could have a "symbol-data" module that would export the `yield all` that way the root-saga wouldn't have to know anything about the api or the concrete actions...